src/runtime/runtime_rpi3.go

```diff
@@ -0,0 +1,120 @@
+// +build rpi3
+package runtime
+
+import (
+	dev "device/arm64"
+	"machine"
+	"runtime/volatile"
+	"unsafe"
+)
+
+const tickMicros = int64(1)
+
+type timeUnit int64
+
+//go:export sleepticks sleepticks
+func sleepTicks(n timeUnit) {
+	WaitMuSec(uint32(n))
+}
+
+func ticks() timeUnit {
+	return timeUnit(SysTimer())
+}
+
+func preinit() {
+	// Initialize .bss: zero-initialized global variables.
+	ptr := unsafe.Pointer(&_start_bss)
+	for ptr != unsafe.Pointer(&_end_bss) {
+		*(*uint32)(ptr) = 0
+		ptr = unsafe.Pointer(uintptr(ptr) + 4)
+	}
+
+	machine.MiniUART.Configure(machine.UARTConfig{})
+}
+
+// main on bare metal (bootloader uses this path)
+//go:export main
+func main() {
+	preinit()
+	primaryMain()
+}
+
+func primaryMain() {
+	initAll()
+	callMain()
+	dev.Abort()
+}
+
+//go:extern
+func putchar(c byte) {
+	//MiniUARTSend(c) if you prefer the mini uart
+	machine.MiniUART.WriteByte(c)
+}
+
+// wait a given number of CPU cycles (at least)
+func WaitCycles(n int) {
+	for n > 0 {
+		dev.Asm("nop")
+		n--
+	}
+}
+
+//
+// Wait MuSec waits for at least n musecs based on the system timer.
+//
+func WaitMuSec(n uint32) {
```

Comment:
I would call this `waitMicroseconds`. Slightly longer but immediately clear. Also, this function is not normally part of the runtime so shouldn't be exported: the normal way if sleeping is `time.Sleep`.